llvm/lib/Target/RISCV/RISCVGatherScatterLowering.cpp

```diff
@@ -349,6 +349,22 @@ RISCVGatherScatterLowering::determineBaseAndStride(Instruction *Ptr,
 
   SmallVector<Value *, 2> Ops(GEP->operands());
 
+  // If the base pointer is a vector, check if it's strided.
```

Comment:
Please move this down into the  if (!ScalarBase) case below, we'd rather find a splatable base if possible.

Please perform the operand check before doing the recursive call.  The unwind from the recursion may generate code.

Reply:
I tried moving this down into the !ScalarBase case but it caused us to miss the splat_base_scalar_offset test case.

The code below bails if GEP doesn't have a vector offset, so putting it up here catches the case where we have a splat base + scalar offset.